src/RoomClient.ts

```diff
@@ -223,6 +277,17 @@ class RoomClient {
       this.emitter.emit("onTimer", { name, msRemaining, msElapsed });
     });
 
+    client.on("peerConnectionState", ({ from, quality, ping }) => {
+      const peer = this.peers[from.id];
+      if (peer) {
+        peer.connectionState.quality = quality;
+        peer.connectionState.ping = ping;
+        // clear events on every state change
+        if (peer.connectionState.events) peer.connectionState.events.length = 0;
```

Comment:
this ... works?

wow apparently this is one of the accepted conventions. thanks javascript.

how about replacing with a new array?

```suggestion
        peer.connectionState.events = []
```